torchtune/datasets/multimodal/_the_cauldron.py

```diff
@@ -235,3 +243,73 @@ def __call__(self, sample: Mapping[str, Any]) -> Mapping[str, Any]:
     if packed:
         raise ValueError("Multimodal datasets don't support packing yet.")
     return ds
+
+
+def the_cauldron_dataset_torchdata(
```

Comment:
How much of this could be pushed into the SFT class so that we can just reuse it for any new datasets?

Reply:
A lot of it is probably re-usable, I think it should be it's own class or at least a builder, maybe like `_sft.py: class SFTDatasetNode` or something less terrible sounding

Reply:
Actually 90% of this builder func could probably live in _sft.py as it doesn't have anything to do with the cauldron